controller/contour/namespace.go

```diff
@@ -50,21 +50,45 @@ func (r *Reconciler) ensureNamespace(ctx context.Context, name string) error {
 	return fmt.Errorf("failed to get namespace %s: %w", ns.Name, err)
 }
 
-// ensureNamespaceRemoved ensures the namespace for the provided name
-// does not exist.
-func (r *Reconciler) ensureNamespaceRemoved(ctx context.Context, name string) error {
-	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: name}}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: ns.Name}, ns)
-	switch {
-	case err == nil:
-		if err := r.Client.Delete(ctx, ns); err != nil {
-			return fmt.Errorf("failed to delete namespace %s: %w", ns.Name, err)
-		}
-		r.Log.Info("deleted namespace", "name", ns.Name)
-		return nil
-	case errors.IsNotFound(err):
-		r.Log.Info("namespace does not exist; skipping removal", "name", ns.Name)
+// ensureNamespaceRemoved ensures the namespace for the provided contour is removed,
+// bypassing deletion if any of the following conditions apply:
+//   - RemoveOnDeletion is set.
```

Comment:
should be "RemoveOnDeletion is **not** set", or "...set to false".

Reply:
ack